torch/nn/parallel/distributed.py

```diff
@@ -173,6 +175,76 @@ def backward(ctx, *grad_outputs):
 
         return (None, None, *grad_outputs)
 
+
+class _DDPJoinHook(_JoinHook):
+    def __init__(self, ddp, divide_by_initial_world_size, enable, throw_on_early_termination):
+        """
+        Sets config variables for internal usage.
+        """
+        assert isinstance(ddp, DistributedDataParallel), \
+            "DDP join hook requires passing in a DistributedDataParallel " \
+            "instance as the state"
+        ddp.logger._set_uneven_input_join()
+        ddp.ddp_uneven_inputs_config = \
```

Comment:
Do we still need this? This doesn't seem used anywhere?

Reply:
Yes, I think we still need this since the behavior of DDP in the forward pass changes depending on all three of these variables (e.g. should the process perform that all-reduce to signal that it has not yet joined). The specific locations are [here](https://github.com/andwgu/pytorch/blob/2c17d363fab22df34cd08b5404758d4adc3a82cf/torch/nn/parallel/distributed.py#L894), [here](https://github.com/andwgu/pytorch/blob/2c17d363fab22df34cd08b5404758d4adc3a82cf/torch/nn/parallel/distributed.py#L897), [here](https://github.com/andwgu/pytorch/blob/2c17d363fab22df34cd08b5404758d4adc3a82cf/torch/nn/parallel/distributed.py#L914), and [here](https://github.com/andwgu/pytorch/blob/2c17d363fab22df34cd08b5404758d4adc3a82cf/torch/nn/parallel/distributed.py#L927).